WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailViewController.swift

```diff
@@ -1031,27 +1032,35 @@ private extension ReaderDetailViewController {
 private extension ReaderDetailViewController {
 
     func configureNavigationBar() {
-
         // If a Related post fails to load, disable the More and Share buttons as they won't do anything.
         let rightItems = [
             moreButtonItem(enabled: enableRightBarButtons),
             shareButtonItem(enabled: enableRightBarButtons),
             safariButtonItem()
         ]
+        navigationItem.largeTitleDisplayMode = .never
+        navigationItem.rightBarButtonItems = rightItems.compactMap({ $0 })
+    }
 
-        if !isModal() {
+    /// Updates the left bar button item based on the current view controller's context in the navigation stack.
+    /// If the view controller is presented modally and does not have a left bar button item, a dismiss button is set.
+    /// If the view controller is not the root of the navigation stack, a back button is set.
+    /// Otherwise, the left bar button item is cleared.
+    func updateLeftBarButtonItem() {
+        if isModal(), navigationItem.leftBarButtonItem == nil {
+            navigationItem.leftBarButtonItem = dismissButtonItem()
+        } else if navigationController?.viewControllers.first !== self {
             navigationItem.leftBarButtonItem = backButtonItem()
         } else {
-            navigationItem.leftBarButtonItem = dismissButtonItem()
+            navigationItem.leftBarButtonItem = nil
         }
-        navigationItem.largeTitleDisplayMode = .never
-        navigationItem.rightBarButtonItems = rightItems.compactMap({ $0 })
     }
 
     func backButtonItem() -> UIBarButtonItem {
-        let button = barButtonItem(with: .gridicon(.chevronLeft), action: #selector(didTapBackButton(_:)))
+        let config = UIImage.SymbolConfiguration(weight: .semibold)
+        let image = UIImage(systemName: "chevron.backward", withConfiguration: config) ?? .gridicon(.chevronLeft)
+        let button = barButtonItem(with: image, action: #selector(didTapBackButton(_:)))
         button.accessibilityLabel = Strings.backButtonAccessibilityLabel
```

Comment:
This change isn't part of the fix, but I've updated the custom back button item to closely match the system back button item.